prov/efa/src/efa_domain.c

```diff
@@ -228,9 +230,50 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 	domain->util_domain.mr_map.mode &= ~FI_MR_PROV_KEY;
 	domain->fab = fabric;
 
+	domain->util_domain.domain_fid.mr = &efa_domain_mr_ops;
+
 	*domain_fid = &domain->util_domain.domain_fid;
 
 	if (efa_mr_cache_enable) {
+		/*
+		 * Register a temporary buffer and call ibv_fork_init() to
+		 * determine if fork support is enabled.
+		 *
+		 * libibverbs maintains a global variable to determine if any
+		 * registrations have occurred before ibv_fork_init() is called.
+		 * We can take advantage of this to determine if
+		 * ibv_fork_init() was called before Libfabric was initialized.
+		 *
+		 * As a temporary work around, EFA will return an error when
+		 * when the cache is enabled and fork support is enabled.
+		 * Using libibverbs fork support and deregistering regions
+		 * after they are unmapped has known bugs; deregistering
+		 * regions after they are unmapped is required for Libfabric
+		 * memory monitor intercept functions to work.
```

Comment:
I think I understand the first half statement, which is because the RBTree is not updated correctly in rdma-core if doing de-registration after the regions are unmapped in fork support enabled case. 

But not quite follow `deregistering regions after they are unmapped is required for Libfabric memory monitor intercept functions to work`. I thought we still want the immediately flush for memhooks monitor, there's just an issue in memhooks interception implementation recently merged. Could you clarify further? Sorry for missing anything.

Reply:
We won't be able to fix the recently discovered issue with memhooks interception.

I'll remove this so there isn't confusion about it in the future.